### PR TITLE
Add sub-resource name when available to script editor

### LINKED ISSUE #24695: Script editor should show names of sub-resources when available
**Godot version:**
Godot 3.1alpha4

In the left panel of the script editor, sub-resource names should be shown when available instead of IDs.  I like not having to create files for scripts used only once but it makes it very difficult to switch between scripts when the names are Game.tscn::1, Game.tscn::2, etc.  I've noticed that you can give sub-resources names, but this is not reflected in the script editor.

What is shown
what is shown

Mockup of what should be shown
mockup of what should be shown

### LINKED ISSUE #28615: Make built-in scripts show the name of the attached node in the script tab
Sort of a revival of #14577

**Godot version:**
3.1.1

**Issue description:**
Working with many built-in scripts in the same scene is really annoying, because the names are 
not very descriptive:

image

Maybe this could be solved by naming the scripts by the name of the node they are attached to.

**Steps to reproduce:**
Create a scene with a bunch of nodes and attach built-in scripts to all of them. No try to find the a script of one node.

Code responsible:
https://github.com/godotengine/godot/blob/b0931806c2a085e84ecc4d7989436edc83cd70a8/editor/plugins/script_editor_plugin.cpp#L1606

### PR DESCRIPTION
**Pull request by uzimonkey**
Fixes #24695 missing sub-resource names in script editor

*Bugsquad edit:* Fixes #28615.

### CODE REVIEW

**Comment by Jummit** [comment]
I think this will close #28615 as well.

**Comment by extrasyrup** [comment]
So, what's the consensus on this one?

File: editor/plugins/script_editor_plugin.cpp [hunk 1]
```diff
@@ -1684,7 +1684,19 @@ void ScriptEditor::_update_script_names() {
 			Ref<Texture> icon = se->get_icon();
 			String path = se->get_edited_resource()->get_path();
 			bool built_in = !path.is_resource_file();
-			String name = built_in ? path.get_file() : se->get_name();
+			String name;
+
+			if (built_in) {
+
+				name = path.get_file();
+				String resourceName = se->get_edited_resource()->get_name();
```
Comment (akien-mga) [inline editor/plugins/script_editor_plugin.cpp]: Please use `snake_case` for identifiers, so this should be `resource_name`.

**Review by akien-mga — APPROVED** [review]
Sorry for the late review. Looks good, just needs a style fix.

File: editor/plugins/script_editor_plugin.cpp (hunk 1 above)
Comment (uzimonkey) [reply editor/plugins/script_editor_plugin.cpp]: I've updated this to use snake case.

**Comment by akien-mga** [comment]
Looks good, could you squash commits together?

**Comment by uzimonkey** [comment]
I've squashed the commits.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**